pkg/commands/git.go

```diff
@@ -471,11 +474,12 @@ func (c *GitCommand) RebaseMode() (string, error) {
 func (c *GitCommand) DiscardAllFileChanges(file *File) error {
 	// if the file isn't tracked, we assume you want to delete it
 	quotedFileName := c.OSCommand.Quote(file.Name)
-	if file.HasStagedChanges {
+	if file.NeedReset {
```

Comment:
i.e. we could do this here: `if file.HasMergeConflicts || file.HasInlineMergeConflicts || file.HasStagedChanges {`

Reply:
this was my original implementation, I changed it to use `NeedReset` just for readability, but I see your point. I'll push a change.